tests/ballerina-test-utils/src/main/java/org/ballerinalang/test/BRunUtil.java

```diff
@@ -1334,10 +1342,13 @@ public static ExitDetails run(CompileResult compileResult, String... args) {
         try {
             final List<String> actualArgs = new ArrayList<>();
             actualArgs.add(0, "java");
-            actualArgs.add(1, "-cp");
+            for (int indexNum = 0; indexNum < javaOpts.size(); indexNum++) {
+                actualArgs.add(indexNum + 1, javaOpts.get(indexNum));
+            }
+            actualArgs.add(javaOpts.size() + 1, "-cp");
             String classPath = System.getProperty("java.class.path") + ":" + getClassPath(classLoader);
-            actualArgs.add(2, classPath);
-            actualArgs.add(3, initClassName);
+            actualArgs.add(javaOpts.size() + 2, classPath);
+            actualArgs.add(javaOpts.size() + 3, initClassName);
             actualArgs.addAll(Arrays.asList(args));
```

Comment:
As mentioned in the previous comment we can continue to use `index++` here, instead of messing with using `javaOpts.size()`

Reply:
I will look into this.